aten/src/ATen/native/quantized/cpu/qembeddingbag_prepack.cpp

```diff
@@ -14,21 +14,35 @@ torch::class_<EmbeddingPackedParamsBase> register_embedding_params();
  * To prepack the weights we store the scale and bias (where bias is Xmin)
  * for each row along with the quantized weights.
  */
-// TODO: Extend this to support 4-bits once 4-bit qtensor support is added.
 c10::intrusive_ptr<EmbeddingPackedParamsBase> PackedEmbeddingBagWeight::prepack(
     at::Tensor qweight) {
   static constexpr int64_t version = 1;
   TORCH_CHECK(
       qweight.dim() == 2,
       "quantized::embedding_bag_prepack weight tensor rank should be 2");
   TORCH_CHECK(
-      qweight.scalar_type() == c10::kQUInt8,
-      "qembedding_bag_prepack currently only supports quint8 weights");
+      qweight.scalar_type() == c10::kQUInt8 ||
+          qweight.scalar_type() == c10::kQUInt4x2,
+      "qembedding_bag_prepack currently only supports quint8 and quint4x2 weights");
 
   at::Tensor weight_contig =
       qweight.contiguous(qweight.suggest_memory_format());
-  const uint8_t* weight_data =
-      reinterpret_cast<uint8_t*>(weight_contig.data_ptr<c10::quint8>());
+
+  uint8_t* weight_data;
+  int bit_width, scale_bias_bytes;
+  if (qweight.scalar_type() == c10::kQUInt8) {
+    weight_data =
+        reinterpret_cast<uint8_t*>(weight_contig.data_ptr<c10::quint8>());
```

Comment:
I remember data_ptr does not do type check, so maybe you can just do weight_contig.data_ptr<uint8_t*>

Reply:
I don't think it is applicable for qint types. It throws the error "expected scalar type Byte but found QUInt8"

Reply:
oh sorry, I meant this: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/templates/TensorBody.h#L354
so:
```
static_cast<uint8_t*>(weight_contig.data_ptr())
```